[schema registry avro] Add Logical Type for DateType (#27318)
Azure/azure-sdk-for-js: sdk/schemaregistry/schema-registry-avro/CHANGELOG.md

# Release History

> ## 1.0.1-beta.1 (Unreleased)

Reviewer: Did you mean 1.1.0-beta.1? Also does this have to be a beta? Are we expecting customer feedback on it?
```suggestion
## 1.1.0-beta.1 (Unreleased)
```

Reviewer: Actually the client is a dep and it is in beta, so you're right this should be in beta as well.

Reviewer: Unless we can depend on the GA version of the client which I think we can. Is there anything the avro serializer needs from the beta client?

Author: The serializer should work with the GA version of the client, I'll test it

Author: I'm currently updating this as a beta version. I'll test the edge cases to see if it's breaking anything for a GA version

### Features Added

> - The Avro Serializer supports serialization and deserialization for `timestamp-millis` logical types to native `Date` types

Reviewer: nitpick:
```suggestion
- Adds support for serializing and deserializing for `timestamp-millis` logical type as native `Date` objects.
```
